airflow/models/dag.py

```diff
@@ -2680,7 +2680,9 @@ class InletRef(NamedTuple):
         outlet_datasets: Dict[Dataset, None] = {}
         input_datasets: Dict[Dataset, None] = {}
         for dag in dags:
-            for dataset in dag.dataset_triggers:
+            if not isinstance(dag.timetable, DatasetTriggeredTimetable):
+                continue
```

Comment:
i think exiting here means you don't iterate through outlets to get outlet datasets

Reply:
Whoops yep.